pkg/services/extsvcauth/oauthserver/errors.go

```diff
@@ -25,3 +26,8 @@ func ErrClientNotFound(clientID string) error {
 		errutil.WithPublicMessage(fmt.Sprintf("Client '%s' not found", clientID))).
 		Errorf("client '%s' not found", clientID)
 }
+
+func IsErrClientNotFound(err error) bool {
+	var srcError errutil.Error
+	return errors.As(err, &srcError) && srcError.MessageID == ErrClientNotFoundMessageID
```

Comment:
Unrelated to this PR but I think this is a bit convoluted. My suggestion would be to define  `ErrClientNotFound` as  a `errutil.Base` error.  And then use it like: 

```
return ErrClientNotFound.Errorf("client '%s' not found", clientID)
```

You'd lose the `clientID` in the public message but the comparison is easier (you can do `errors.Is(err, ErrClientNotFound)`

Reply:
I looked into it a bit today thinking this was convoluted as well and I think it's done this way to be able to have the clientID in the public message :thinking:

Reply:
It can bubble up from fosite when requesting a token. But on the other hand fosite must wrap it anyway :thinking:

Reply:
I think I'll change it!
But I'll do it in another PR, I don't want to have to run through CI again :D